zarr_etl_tools/utils/store.py

```diff
@@ -178,6 +181,92 @@ def has_existing(self) -> bool:
         """
         return self.fs().exists(self.url)
 
+    def push_metadata(self, title: str, stac_content: dict, stac_type: str):
+        """
+        Publish metadata entity to s3 store. Tracks historical state
+        of metadata as well
+
+        Parameters
+        ----------
+        title : str
+            STAC Entity title
+        stac_content : dict
+            content of the stac entity
+        stac_type : StacType
+            Path part corresponding to type of STAC entity
+            (empty string for Catalog, 'collections' for Collection or 'datasets' for Item)
+        """
+        metadata_path = self.get_metadata_path(title, stac_type)
+        if self.fs().exists(metadata_path):
+            # Generate history file
+            old_mod_time = self.fs().ls(metadata_path, detail=True)[0]["LastModified"]
+            history_path = f"s3://{self.bucket}/history/{title}/{old_mod_time.isoformat(sep='T')}.json"
```

Comment:
Wouldn’t it be better to include the dataset name in the history file? Then if we download it / pull it it’ll be easier to keep track of.